### PR TITLE
Align AGNSS CDDL with cloud and add support for AGNSS appid and topic 

### PR DESCRIPTION
**Pull request by plskeggs**
The cloud side simplified the CDDL for AGNSS compared to AGPS, which changed the ID for TAC. Make the corresponding changes here.

Complete the implementation of support for the AGNSS appid, which replaces the AGPS appid.

On the nrf91x1 series, this includes support for QZSS assistance where available geographically.

Jira: IRIS-7294

test-sdk-nrf:sdk-nrf-pr-13468

### CODE REVIEW

**Review by jayteemo — APPROVED** [review]

**Review by glarsennordic — APPROVED** [review]

**Comment by plskeggs** [comment]
Note: this will fail test-fw-nrfconnect-nrf-iot_cloud until this test PR is merged: https://projecttools.nordicsemi.no/bitbucket/projects/NCS-TEST/repos/test-fw-nrfconnect-nrf_iot/pull-requests/4144/

**Comment by glarsennordic** [comment]
I have manually run CI tests against this PR with the updates from https://projecttools.nordicsemi.no/bitbucket/projects/NCS-TEST/repos/test-fw-nrfconnect-nrf_iot/pull-requests/4144/ , and they passed. Results are here: https://jenkins-ncs.nordicsemi.no/job/latest/job/test-fw-nrfconnect-nrf-iot_cloud/job/PR-4144/5/consoleFull

We should be good to merge both, despite CI looking red on github

**Comment by trantanen** [comment]
> I have manually run CI tests against this PR with the updates from https://projecttools.nordicsemi.no/bitbucket/projects/NCS-TEST/repos/test-fw-nrfconnect-nrf_iot/pull-requests/4144/ , and they passed. Results are here: https://jenkins-ncs.nordicsemi.no/job/latest/job/test-fw-nrfconnect-nrf-iot_cloud/job/PR-4144/5/consoleFull
> 
> We should be good to merge both, despite CI looking red on github

You can add test PR info into this PR and then the CI will use it and pass. So you can get the CI to green on this one. I'll send you instructions privately.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -626,6 +626,7 @@ Libraries for networking
 
     * The :c:func:`nrf_cloud_coap_shadow_delta_process` function to include a parameter for application-specific shadow data.
     * The :c:func:`nrf_cloud_coap_shadow_delta_process` function to process default shadow data added by nRF Cloud, which is not used by CoAP.
+    * The CDDL file for AGNSS to align with cloud changes, and regenerated the AGNSS encoder accordingly.
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:629]:
```suggestion
    * The CDDL file for AGNSS to align with cloud changes and regenerated the AGNSS encoder accordingly.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by trantanen — APPROVED** [review]

**Review by peknis — APPROVED** [review]

File: subsys/net/lib/nrf_cloud/coap/cddl/nrf_cloud_coap_agnss.cddl [hunk 2]
```diff
@@ -28,11 +25,8 @@ filtered = 3
 mask = 4
 mcc = 5
 mnc = 6
-requestType = 7
-rsrp = 8
-tac = 9
-rtAssistance = 10
-custom = 11
+rsrp = 7
```
Comment (tokangas) [inline subsys/net/lib/nrf_cloud/coap/cddl/nrf_cloud_coap_agnss.cddl:28]: This changes the IDs for `rsrp` and `tac`. Does this mean that there has been backwards incompatible changes to the cloud API?

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_fsm.c [hunk 3]
```diff
@@ -701,6 +701,7 @@ static int dc_rx_data_handler(const struct nct_evt *nct_evt)
 
 	switch (nrf_cloud_dc_rx_topic_decode(cloud_evt.topic.ptr)) {
 	case NRF_CLOUD_RCV_TOPIC_AGPS:
```
Comment (tokangas) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud_fsm.c:703]: Is it necessary to keep the AGPS case, defines etc.? I would think that nothing is received on the AGPS topic if the device does not send a request for it.

File: subsys/net/lib/nrf_cloud/coap/cddl/nrf_cloud_coap_agnss.cddl (hunk 2 above)
Comment (plskeggs) [reply subsys/net/lib/nrf_cloud/coap/cddl/nrf_cloud_coap_agnss.cddl:28]: Numbers were changed only on CoAP's agnss endpoint, not for the existing apgs endpoint, so older code should continue to work.

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_fsm.c (hunk 3 above)
Comment (plskeggs) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud_fsm.c:703]: Good catch, thanks.

**Review by tokangas — APPROVED** [review]

**Pull request merged**